core/server/worker/src/main/java/alluxio/worker/block/io/DeStoreBlockReader.java

```diff
@@ -0,0 +1,62 @@
+package alluxio.worker.block.io;
+
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+
+import io.netty.buffer.ByteBuf;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+
+/**
+ * An reader class with metrics.
+ */
+public class DeStoreBlockReader extends BlockReader {
+  private final BlockReader mDeBlockReader;
+
+  /**
+   * A decorator of BlockReader.
+   * @param deblockReader block reader
+   */
+  public DeStoreBlockReader(BlockReader deblockReader) {
+    mDeBlockReader = deblockReader;
+  }
+
+  @Override
+  public ByteBuffer read(long offset, long length) throws IOException {
+    return mDeBlockReader.read(offset, length);
+  }
+
+  @Override
+  public long getLength() {
+    return mDeBlockReader.getLength();
+  }
+
+  @Override
+  public ReadableByteChannel getChannel() {
+    return mDeBlockReader.getChannel();
```

Comment:
the channel also needs to be wrapped to add the metric updating logic.
something looks like

```java
return new ReadableByteChannel() {
  private final ReadableByteChannel mDelegate = mDeBlockReader.getChannel();

  @Override
  public int read(ByteBuffer dst) throws IOException {
    int bytesRead = mDelegate.read(dst);
    // update metric here
    return bytesRead;
  }
}
```